extension-iap/src/iap_ios.mm

```diff
@@ -557,6 +563,9 @@ static void IAP_OnCommand(IAPCommand* cmd, void*)
 static dmExtension::Result UpdateIAP(dmExtension::Params* params)
 {
     IAP_Queue_Flush(&g_IAP.m_CommandQueue, IAP_OnCommand, 0);
+    if (g_IAP.m_Observer) {
+        IAP_Queue_Flush(&g_IAP.m_ObservableQueue, IAP_OnCommand, 0);
```

Comment:
This extra queue isn't needed?
What's the rationale behind it?
I would assume pushing the commands to the `m_CommandQueue` would suffice.

Reply:
There is another way to solve this: put to the queue command with empty callback and check if its empty on queue processing, and when listener setup iterate queue and setup callback for empty callbacks but I think its worse.

Reply:
Now that I understand what this queue is for, and how the code flow is, I think this PR is a good way of solving it.